docs/runbooks/KubeVirtVMHighCPUUsage.md

```diff
@@ -0,0 +1,144 @@
+# KubeVirtVMHighCPUUsage
+
+## Meaning
+
+This alert fires when a Virtual Machine (VM) has been consuming more than 80% of its allocated CPU resources for more than 5 minutes.
+
+## Impact
+
+High CPU usage in a VM can indicate:
+- **Performance degradation** - Applications running inside the VM may become slow or unresponsive
+- **Resource contention** - The VM may be competing for CPU resources with other workloads
+- **Potential application issues** - Runaway processes, infinite loops, or inefficient code
+- **Capacity planning needs** - The VM may need additional CPU resources
+
+## Diagnosis
+
+### 1. Identify the Affected VM
+
+The alert provides the VM Name, Namespace, Node and CPU Usage.
+
+### 2. Check VM CPU Allocation
+
+```bash
+# Get VM configuration
+kubectl get vm <vm-name> -n <namespace> -o yaml
+
+# Check current VMI status
+kubectl get vmi <vm-name> -n <namespace> -o yaml
+```
+
+Look for:
+- `spec.domain.cpu.cores` - Number of allocated vCPUs
+- `spec.domain.resources.requests.cpu` - CPU resource requests
+- `spec.domain.resources.limits.cpu` - CPU resource limits
+
+### 3. Monitor Current CPU Usage
+
+```bash
+# Check real-time CPU metrics
+kubectl exec -n <namespace> $(kubectl get pods -n <namespace> -l kubevirt.io/vmi=<vm-name> -o jsonpath='{.items[0].metadata.name}') -- top
+
+# Check VM resource consumption
+virtctl top vmi <vm-name> -n <namespace>
+```
+
+### 4. Check for Processes Inside the VM
+
+Connect to the VM to identify high CPU processes:
+
+```bash
+# Console access
+virtctl console <vm-name> -n <namespace>
+
+# Or SSH if available
+virtctl ssh <vm-name> -n <namespace>
+```
+
+Inside the VM, run:
+```bash
+# Top processes by CPU usage
+top -c
+
+# Process tree
+htop
+
+# Detailed process information
+ps aux --sort=-%cpu | head -20
+```
+
+### 5. Check Node-Level Metrics
+
+```bash
+# Check node resource usage
+kubectl top node <node>
+
+# Check other VMs on the same node
+kubectl get vmi --all-namespaces --field-selector spec.nodeName=<node>
+
+# Check node conditions
+kubectl describe node <node>
+```
+
+## Mitigation
+
+### Immediate Actions
+
+1. **Scale VM Resources** (if possible):
+   ```bash
+   # Edit VM to increase CPU allocation
+   kubectl patch vm <vm-name> -n <namespace> --type='merge' -p='{"spec":{"template":{"spec":{"domain":{"cpu":{"cores":4}}}}}}'
```

Comment:
What if instancetypes are used?

Reply:
What do you suggest to add here?

Reply:
Make it more generic, mention editing the VM's spec or choosing a different instancetype.